src/main/java/com/icthh/xm/ms/balance/service/BalanceService.java

```diff
@@ -269,6 +277,216 @@ private Optional<Pocket> findPocketForReload(ReloadBalanceRequest reloadRequest,
         return pocket;
     }
 
+    @Transactional
+    public BalanceChangeEventDto revertBalanceOperation(RevertBalanceOperationRequest revertRequest) {
+        log.info("Start revert balance operation with request {}", revertRequest);
+        Balance balance = getBalanceForUpdate(revertRequest.getBalanceId());
+        return self.revertBalanceOperation(balance, revertRequest);
+    }
+
+    @Transactional
+    @LogicExtensionPoint(value = "Revert", resolver = BalanceTypeKeyResolver.class)
+    public BalanceChangeEventDto revertBalanceOperation(Balance balance, RevertBalanceOperationRequest revertRequest) {
+        String operationUuid = revertRequest.getUuid();
+
+        Optional<BalanceChangeEventDto> duplicateOperation = findExistBalanceChangeEventDto(operationUuid, false);
+        if (duplicateOperation.isPresent()) {
+            return duplicateOperation.get();
+        }
+
+        log.info("Start revert balance operation with UUID: {}", revertRequest.getUuidToRevert());
+
+        BalanceChangeEventDto operationToRevert = findExistBalanceChangeEventDto(revertRequest.getUuidToRevert(), true)
+            .orElseThrow(() -> {
+                log.error("Not found balance change event with UUID: {}", revertRequest.getUuidToRevert());
+                return new EntityNotFoundException("Balance change event with UUID " + revertRequest.getUuidToRevert() + "not found");
+            });
+
+        boolean isRevertOperationAlreadyProvided = balanceChangeEventRepository.existsByRevertOperationId(revertRequest.getUuidToRevert());
+        if (isRevertOperationAlreadyProvided) {
+            log.error("Revert operation for UUID: {} - already provided", revertRequest.getUuidToRevert());
+            throw new BusinessException("error.revert.operation.already.provided", "Revert operation already provided");
+        }
+
+        OperationType previousOperationType = operationToRevert.getOperationType();
+        log.info("Operation type to revert: {}", previousOperationType);
+        if (previousOperationType.equals(RELOAD)) {
+            return revertReloadOperation(balance, revertRequest, operationToRevert);
+        } else if (previousOperationType.equals(CHARGING)) {
+            return revertChargingOperation(balance, revertRequest, operationToRevert);
+        } else {
+            log.error("Could not operation provide, because revert operation type: {} - not supported", operationToRevert.getOperationType());
```

Comment:
`Could not operation provide, because revert operation type: {} - not supported` -> `Revert of {} operation type is not supported`

Reply:
Fixed